react-app/components/FileManager/index.jsx

```diff
@@ -34,8 +34,11 @@ const FileManager = (props) => {
   const [, setIsFetchLoading] = useState(false);
   const [downloadFileErrorMessage, setDownloadFileErrorMessage] = useState(null);
   const [reloadPage, setReloadPage] = useState(false);
+  const downloadFilesData = useRef(null);
 
   const onFileDownloaded = () => {
+    const contents = downloadFilesData.current;
+    const objectData = getMetadataFiles(contents);
     setDownloadFileErrorMessage(null);
```

Comment:
`in constants/xBlockContext.js`

xBlockContext add these propierties

courseId: null,
userId: null
username:  null

then in 
`react-app/index.js`

xBlockContext.xblockId = context.xblock_id
xBlockContext.courseId = context.user_id
xBlockContext.userId   =  context.user_id
xBlockContext.username = context.username

After that you could use them here

```
const { isStudioView, xblockId, courseId, userId, username } = xBlockContext;

if(!isStudioView) {
      sendTrackingLogEvent('edunext.xblock.filesmanager.files.downloaded', {
        course_id: courseId,
        xblock_id: xblockId,
        user_id: userId,
        username: username,
        files_downloaded_metadata: objectData,
        created_at: new Date().toISOString(),
      });
    }
```

Reply:
Thanks!, done